include/eigenpy/std-array.hpp

```diff
@@ -0,0 +1,140 @@
+/// Copyright (c) 2023 CNRS INRIA
+
+#ifndef __eigenpy_utils_std_array_hpp__
+#define __eigenpy_utils_std_array_hpp__
+
+#include <boost/python/suite/indexing/indexing_suite.hpp>
+#include "eigenpy/std-vector.hpp"
+
+#include <array>
+
+namespace eigenpy {
+
+template <typename Container, bool NoProxy, class SliceAllocator,
+          class DerivedPolicies>
+class array_indexing_suite;
+namespace details {
+
+template <typename Container, bool NoProxy, class SliceAllocator>
+class final_array_derived_policies
+    : public array_indexing_suite<
+          Container, NoProxy, SliceAllocator,
+          final_array_derived_policies<Container, NoProxy, SliceAllocator> > {};
+}  // namespace details
+
+template <typename Container, bool NoProxy = false,
+          class SliceAllocator = std::allocator<typename Container::value_type>,
+          class DerivedPolicies = details::final_array_derived_policies<
+              Container, NoProxy, SliceAllocator> >
+class array_indexing_suite
+    : public bp::vector_indexing_suite<Container, NoProxy, DerivedPolicies> {
+ public:
+  typedef typename Container::value_type data_type;
+  typedef typename Container::value_type key_type;
+  typedef typename Container::size_type index_type;
+  typedef typename Container::size_type size_type;
+  typedef typename Container::difference_type difference_type;
+  typedef std::vector<data_type, SliceAllocator> slice_vector_type;
+  static constexpr std::size_t Size = std::tuple_size<Container>{};
+
+  template <class Class>
+  static void extension_def(Class &) {}
+
+  template <class Iter>
+  static void extend(Container &, Iter, Iter) {}
+
+  static void append(Container &, data_type const &) {}
+  // no-op
+  static void delete_item(Container &, index_type) {}
+  // no-op
+  // no-op
+  static void delete_slice(Container &, index_type, index_type) {}
+
+  static void set_slice(Container &container, index_type from, index_type to,
+                        data_type const &v) {
+    if (from > to) {
+      return;
+    } else {
+      std::fill(container.begin() + from, container.begin() + to, v);
+    }
+  }
+
+  template <class Iter>
+  static void set_slice(Container &container, index_type from, index_type to,
+                        Iter first, Iter last) {
+    if (from > to) {
+      return;
+    } else {
+      std::copy(first, last, container.begin() + from);
+    }
+  }
```

Comment:
`set_slice` behavior in python is not exactly the same.

Elements between `from` and `to` are deleted. This mean, the container can be resized, and that not possible for std::array.

On the first `set_slice`, I think your version make sens. But, it can also be misleading when working in Python.

In the second version, you should ensure that `first`, `last` fit the array size.
The following code in the unit test create an invalid write : `ints[:] = [1, 2, 3, 4]`